orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/CloudOperationStage.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.clouddriver.pipeline;
+
+import static java.lang.String.format;
+
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder;
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageGraphBuilder;
+import com.netflix.spinnaker.orca.api.pipeline.graph.TaskNode;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.clouddriver.model.OperationLifecycle;
+import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorCloudOperationTask;
+import com.netflix.spinnaker.orca.clouddriver.tasks.SubmitCloudOperationTask;
+import com.netflix.spinnaker.orca.kato.pipeline.Nameable;
+import java.util.*;
+import javax.annotation.Nonnull;
+import lombok.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+public class CloudOperationStage implements StageDefinitionBuilder, Nameable {
+
+  private static final String LIFECYCLE_KEY = "operationLifecycle";
+
+  protected TaskGraphConfigurer configureTaskGraph(@Nonnull StageExecution stage) {
+    return new TaskGraphConfigurer();
+  }
```

Comment:
`ResizeServerGroupStage` will work as-is without further configuration of the task graph, but for more complex stages, such as Deploy and so-on, they'll need to extend this stage to decorate the task graph more - this method is forward-looking to those use cases.